cleverhans/attacks.py

```diff
@@ -2456,3 +2423,50 @@ def arg_type(arg_names, kwargs):
     dtypes.append(dtype)
   dtypes = tuple(dtypes)
   return (passed, passed_and_not_none, dtypes)
+
+def optimize_linear(grad, eps, ord=np.inf):
+  """
+  Solves for the optimal input to a linear function under a norm constraint.
+
+  Optimal_perturbation =
+    argmax(eta, norm(eta, ord) < 1) dot(eta, grad)
```

Comment:
1) this function solved the problem subject to constraint ||eta|| <= epsilon (rather than <= 1.0)
2) Does people typically use LaTex notation in the CleverHans comments? It might be easier to read (at least for people familiar with latex):

```
 Optimal_perturbation = argmax_{eta, ||eta||_{ord} < eps} dot(eta, grad)
```

Reply:
Done